charts/seed-operatingsystemconfig/original/templates/containerd/_containerd.service

```diff
@@ -1,10 +0,0 @@
-{{- define "containerd-service" -}}
-- name: containerd.service
-  command: restart
-  enable: true
-  dropIns:
```

Comment:
@nimrodoron does the whole containerd instrumentation still work now? I guess the service was named `containerd.service` on purpose and designed as drop-in?

Reply:
First ContainerD was started with the default configurations of the image and later on it was restarted after the configurations has changed.
There is also important order of startup of ContainerD and Docker, After this change it should be checked that both of them are running ok. 
Finally the ContainerD is running with the default config.toml so there is not problem with Container Runtimes and also docker restarted successfully.

Reply:
By the way, For Ubuntu, the default configuration is similar to the generated config.toml.
The socket path is not changed (Like in CoreOS or SuSe) So the ContainerD works well before and after the restart. Also ContainerD starting well also if the config file is not there with default configurations.

Reply:
Does it mean you are OK with this change?

Reply:
Well, for Ubuntu at the moment it seems ok if it works. But At the end of the initialization ContainerD & Docker worked fine, we both checked, and ContainerD did use the config.toml.
Are you sure there is a bug after the initialization is finished?

For CoreOS and SuSe where the config.toml changing the socket path and docker should be restart after the ContainerD with the new configurations so i'm not sure if this will work well, but currently we didn't push them because of other reasons...

Reply:
I am not quite sure if I understand what you mean here. 
The changes introduced in this PR make sure that
- the config file is generated
- introduces a dependency from the containerd.service to the config-generator.service. That means on each cloud-config update, the generator service will be restarted to generate the config file and then the dependent containerd.service will be restarted as well. 
I think that is the same thing you aimed just realized in a different way
. 
Problem was, that there was a configuration error of the drop-in file for the containerd.service so that the mtu-customizer service was actually placed inside the 10-pre-xxx.conf file & not the desired content.
I introduced another service to have a separation between the generation of the config file and the containerd execution that is logically dependent on it. 

Regarding the socket path change on CoreOS and Suse: the socket path is configured in the config toml as you mentioned. The generator service, consequently also the containerd service should be restarted when the cloud-config changes (by the cloud-config execution script).  Also the PartOf dependecy makes sure that the generation of the config file is complete before containerd is restarted, so that it is configured with the new socket path.


Let me know if that makes sense and of course please test it. Also check that without the PR there are several units in an erroneous state.

Reply:
Ok, sounds pretty much similar. 
Regarding the details in the bug.

I didn't check the mtu-customizer but:
8. execute ls etc/containerd
Observe that there is no config.yaml for containerd.

9. execute systemctl status containerd.service
Observe that it is in an error state due to a drop-in file.

From my checks it worked ok the file created and containerd service worked well.

Reply:
The erroneous drop-in file nor the absence of the config file prevent the containerd process from starting. From the kubelets perspective the containerd socket was reachable and usable under its default socket path.

Reply:
If containerd service is down the socket is not reachable and Kubelet can't work.

I have just created a cluster with Ubuntu.
config.toml was created successfully.
containerd.service status is running successfully.

I don't find mtu-customizer service as you mentioned.

I don't mind if you apply this change if it solving the mtu-customizer, i don't know this service.

Reply:
Ok, talking with @rfranzke i understood the bug is on AWS. I checked on GCP where everything was fine. I will check the fix on GCP.

Reply:
ok makes sense. Apparently the mtu-customizer service is only added on AWS, thats why it has not been observed on GCP. Thanks for checking

Reply:
Ok I checked your change in GCP as well, Works. great :)